### PR TITLE
Issue 61 & Issue 62, #team 2

### PR DESCRIPTION
**Pull request by zupan**
This pull request includes issue 61 and issue 62.

### CODE REVIEW

File: piplmesh/account/forms.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 from django import forms
 from django.conf import settings
-from django.contrib.auth import forms as auth_forms, models as auth_models
+from django.contrib.auth import forms as auth_forms
+from mongoengine.django import auth as auth_models
```
Comment (mitar) [inline piplmesh/account/forms.py:6]: Put mongoengine imports into a separate import group (after Django).

File: piplmesh/account/models.py [hunk 2]
```diff
@@ -1,18 +1,23 @@
-from django.contrib.auth import models as auth_models
-from django.db import models as django_models
+from mongoengine.django import auth as auth_models
```
Comment (mitar) [inline piplmesh/account/models.py]: Put mongoengine imports into a separate import group (after Django). Order alphabetically my module names (so first go import mongoengine and then django support).

File: piplmesh/account/models.py [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
+import mongoengine
 
-class UserProfile(django_models.Model):
+class UserProfile(mongoengine.Document):
     """ 
     Class used for storing additional information about user.
     """
 
-    user = django_models.OneToOneField(auth_models.User)
+    user = mongoengine.ReferenceField(auth_models.User);
```
Comment (mitar) [inline piplmesh/account/models.py:10]: NOOOOOOOOOOO. The whole idea of going to MongoDB for User object is that we will have one object for everything: for user authentication data and for user profile data. All this should be combined into one mongoengine document.

File: piplmesh/account/models.py [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 
     # Custom fields
-    birthdate = django_models.DateField(blank=True, null=True)
-    gender = django_models.CharField(max_length=6, blank=True)
-    facebook_id = django_models.BigIntegerField(verbose_name=u'Facebook ID', null=True, blank=True)
-    token = django_models.CharField(max_length=150)
+    language = mongoengine.StringField(required=True, default="sl");
```
Comment (mitar) [inline piplmesh/account/models.py:13]: Don't hardcode. Or at least move to settings (or use existing settings).

Comment (mitar) [inline piplmesh/account/models.py:13]: For language you have that.

File: piplmesh/account/views.py [hunk 5]
```diff
@@ -20,7 +20,7 @@ def registration_view(request):
     form = forms.RegistrationForm()
     if request.method == 'POST':
         form = forms.RegistrationForm(request.POST)
-        if form.is_valid():
+        if form.is_valid():
```
Comment (mitar) [inline piplmesh/account/views.py:23]: Why additional space?

File: piplmesh/settings.py [hunk 6]
```diff
@@ -183,6 +188,7 @@
 AUTH_PROFILE_MODULE = 'account.UserProfile'
 
 AUTHENTICATION_BACKENDS = (
+    'mongoengine.django.auth.MongoEngineBackend',
```
Comment (mitar) [inline piplmesh/settings.py:177]: No, we want only mongoengine-based authentications. Not also others. We have to rewrite others to work on mongoengine.

File: piplmesh/settings.py [hunk 7]
```diff
@@ -28,6 +28,11 @@
     }
 }
 
+from mongoengine import connect
```
Comment (mitar) [inline piplmesh/settings.py:20]: Good, but you have to throw relation database configuration out. Only then we will be sure that we are not using it somewhere.

**Comment by mitar** [comment]
Please comment when you update the ticket because I do not get notification otherwise.

File: piplmesh/settings.py [hunk 8]
```diff
@@ -183,8 +180,7 @@
 AUTH_PROFILE_MODULE = 'account.UserProfile'
 
 AUTHENTICATION_BACKENDS = (
-    'account.backends.CaseInsensitiveModelBackend',
-    'account.backends.FacebookBackend', 
+    'mongoengine.django.auth.MongoEngineBackend',
```
Comment (mitar) [inline piplmesh/settings.py:177]: Is it case sensitive or insensitive? You could extend and use our own.

File: piplmesh/settings.py [hunk 9]
```diff
@@ -145,7 +142,7 @@
     'piplmesh.frontend',
     'piplmesh.account',
     
-    'django.contrib.auth',
+    #'django.contrib.auth',
```
Comment (mitar) [inline piplmesh/settings.py]: Don't comment out. Remove or leave. We have repository to take care of the rest.

File: piplmesh/settings.py [hunk 10]
```diff
@@ -145,7 +142,7 @@
...
     'django.contrib.contenttypes',
```
Comment (mitar) [inline piplmesh/settings.py:142]: I think also this should go out?

File: piplmesh/settings.py [hunk 11]
```diff
@@ -145,7 +142,7 @@
...
     'django.contrib.sessions',
     'django.contrib.sites',
```
Comment (mitar) [inline piplmesh/settings.py:144]: I think also this should go out?

File: piplmesh/settings.py [hunk 12]
```diff
@@ -183,8 +180,7 @@
 AUTH_PROFILE_MODULE = 'account.UserProfile'
```
Comment (mitar) [inline piplmesh/settings.py]: I am not sure if this works/is needed?

File: piplmesh/settings.py [hunk 13]
```diff
@@ -18,16 +18,13 @@
 MANAGERS = ADMINS
 
 DATABASES = {
```
Comment (mitar) [inline piplmesh/settings.py]: Completely remove. Remove also `database_file` variable.

Comment (mitar) [inline piplmesh/settings.py]: Check what `./manage syncdb` does. ;-)

File: piplmesh/account/views.py (hunk 5 above)
Comment (mitar) [inline piplmesh/account/views.py:23]: TODO?

File: piplmesh/account/models.py (hunk 3 above)
Comment (mitar) [inline piplmesh/account/models.py:10]: TODO?

File: piplmesh/account/backends.py [hunk 14]
```diff
@@ -10,7 +10,7 @@ class CaseInsensitiveModelBackend(backends.ModelBackend):
     """
     This backend uses case-insensitive username authentication which is not supported by default.
     """
-    
+    #TODO: Fix to work with mongo database.
```
Comment (mitar) [inline piplmesh/account/backends.py:13]: Probably this is not needed anymore, depending how does mongodb backend do the work?

**Comment by mitar** [comment]
Please upgrade to the latest from main repository so that automatic merge can be done.

File: piplmesh/settings.py (hunk 13 above)
Comment (maticko) [inline piplmesh/settings.py]: Removed.

File: piplmesh/settings.py (hunk 9 above)
Comment (maticko) [inline piplmesh/settings.py]: Removed.

**Comment by mitar** [comment]
Closing in favor of #102.

**Pull request closed without merging**